src/api/api_numeral.cpp

```diff
@@ -459,33 +459,36 @@ extern "C" {
         Z3_CATCH_RETURN(nullptr);
     }
 
-#if 0
-    Z3_ast Z3_API Z3_mk_mpz_numeral(Z3_context c, bool sign, unsigned n, unsigned const nums[], Z3_sort* srt) {
-        LOG_TRY;
-        LOG_Z3_mk_mpz_numeral(c, sign, n, nums, srt);
+    Z3_ast Z3_API Z3_mk_mpz_numeral(Z3_context c, bool sign, unsigned n, unsigned const nums[], Z3_sort srt) {
+        // LOG_TRY;
+        // LOG_Z3_mk_mpz_numeral(c, sign, n, nums, srt);
         RESET_ERROR_CODE();
-        rational z;
 
-        // todo fill in z
-        if (!z.size())
-            z.neg();
-        arith_util & a = mk_c(c)->autil();
-        auto* a = mk_c(c)->mk_numeral_core(r, a.mk_int_sort());
-        Z3_CATCH_RETURN(nullptr);
+        unsigned den[] = {1};
+        rational value(n, nums, 1, den);
+        if (!sign)
+            value.neg();
```

Comment:
Coersion into digit_t is TBD.
```
#if sizeof(unsigned) == sizeof(digit_t)

// no coercion

#else 
// compile assert sizeof(digit_t) == 8
buffer<unsigned> _num;
unsigned i = 0;
for (; i + 1 < n; i += 2) 
buffer.push_back((digit_t)num[i] + (digit_t))num[i+1] << 32ull);
if (i < n) 
   buffer.push_back(num[i]);

rational value(buffer.size(), buffer.data());
#endif
```

Reply:
The coersion already happens now, but in the lowest level (in `mpz.h` line 438ff).

Reply:
It is not yet part of your pull request.